wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -453,15 +454,18 @@ public void cancel(Command... commands) {
       if (!isScheduled(command)) {
         continue;
       }
-
-      m_scheduledCommands.remove(command);
-      m_requirements.keySet().removeAll(command.getRequirements());
       command.end(true);
       for (Consumer<Command> action : m_interruptActions) {
         action.accept(command);
       }
+      m_scheduledCommands.remove(command);
+      m_requirements.keySet().removeAll(command.getRequirements());
       m_watchdog.addEpoch(command.getName() + ".end(true)");
     }
+    m_deferScheduleAndCancel = false;
+    while (!m_toCancel.isEmpty() && !m_deferScheduleAndCancel) {
+      cancel(m_toCancel.poll());
+    }
```

Comment:
A few notes which I thought I'd mention:
* Really obscure detail: Canceling multiple commands at a time will behave slightly differently between Java and C++, since in Java the recursion happens after processing all commands while in C++ it happens after processing each command, resulting in a different order of processing. This is an edge case that teams or libraries can probably figure out for themselves, but given that this PR is about consistency I thought I'd at least mention it, especially since there's a quick fix- Java could be made to behave like C++ by moving this inside the loop, among other options.
* There will be a recursive call for each element of `m_toCancel`- That is, if we reach this code and `m_toCancel` consists of `cmd1`, `cmd2`, and `cmd3` in that order, we will call `cancel(cmd1)`, which will call `cancel(cmd2)`, which will then call `cancel(cmd3)`. I don't think this warrants any code change except maybe a comment- This \*shouldn't\* be a major performance hit because `m_toCancel` is empty in most typical situations, and keeping it as a while loop is more future proof- but I thought this would be worth having on the radar.

Reply:
Alternatively, instead of recursion, perhaps extract the unscheduling logic to a separate (private) method (akin to `initCommand`)? This method could be used also in other places that unschedule commands.

Reply:
Canceling multiple commands is now consistent; there's now a `cancel` that accepts individual commands in both languages.

As for the recursive calls, I could copy the queue contents to an array, clear the queue, and then call `cancel` with the array, although that would be another allocation. And teams generally shouldn't be calling `cancel` from commands, so for most cases it won't even reach that point. `m_toCancel` only fills when `cancel` is called in the loop or during `cancel`. It's highly unlikely that a team is canceling multiple commands during the loop or `cancel`, so performance should be fine.